src/inmanta/protocol/openapi/model.py

```diff
@@ -68,6 +70,61 @@ class SchemaBase(BaseModel):
     anyOf: Optional[Sequence["SchemaBase"]] = None
     enum: Optional[List[str]] = None
 
+    def from_dict(self: S, schema: Dict[str, Any]) -> S:
```

Comment:
So, this comment got me into an interesting journey.  
First, thanks for the explanation, it would indeed look better with object, the reason I was a bit resilient to do so was to match the types returned by `pydantic.schema.model_schema` that would end up being used here.  

But, while looking for the return types of `model_schema`, I noticed that the values I was actually supposed to pass to this method, which are coming directly from the `Components`'s `schemas` attribute, are not supposed to be dictionaries but `Schema`.  And the reason I got confused and mistyped my method is because the `schemas` attribute is actually misused, filled up with dictionaries instead of schemas.  

After some time trying to fix this elegantly, and a quick discussion with @andraskvr , we decided that the `SchemaBase` object might not have any reason to be, and we merged it with `Schema`, simplifying everything.  And while changing that, I also fixed the usage of `Components.schemas` and the definition of this method.